### PR TITLE
Net writer should not produce nan as fill values

### LINKED ISSUE #290: net writer should not produce NaNs as fill values
**Describe the bug**
When creating  basic 2D mesh and saving it, unsupported NaN values are written for the mesh node z values.
(Discovered in a test by @Hofer-Julian)

**To Reproduce**
1. Execute this script:
```python
from hydrolib.core.io.net.models import NetworkModel

network = NetworkModel()
network.mesh2d_create_rectilinear_within_extent(extent=(-5, -5, 5, 5), dx=1, dy=1)
network.save()
``` 
2. The working dir should now contain a file `network.nc`
3. Inspect that file via a NetCDF viewer (panoply / hdfview / ..)
4. Observe NaNs in:
```
        double mesh2d_node_z(mesh2d_nNodes) ;
                mesh2d_node_z:_FillValue = NaN ;
                mesh2d_node_z:standard_name = "altitude" ;
// ...
        double mesh2d_face_z(mesh2d_nFaces) ;
                mesh2d_face_z:_FillValue = NaN ;
                mesh2d_face_z:units = "m" ;
                mesh2d_face_z:mesh = "mesh2d" ;

```

**Expected behavior**
Use numeric (non-NaN) _FillValue instead, for example the netcdf default value (available in `netCDF4.default_fillvals{'f8'}` for doubles for example.
Find an alternative for current code:
```python
        mesh2d_node_z = ncfile.createVariable(
            "mesh2d_node_z", np.float64, nc_mesh2d.node_dimension, fill_value=np.nan
        )
```

### PR DESCRIPTION
**Pull request by tim-vd-aardweg**
Fixes #290

### CODE REVIEW

File: tests/io/test_net.py [hunk 1]
```diff
@@ -559,3 +562,46 @@ def test_add_1d2d_links():
...
+def test_write_netcdf_with_custom_fillvalue_correctly_writes_fillvalue():
+    nc_output_file = Path(test_output_dir / "test.nc")
+
+    # create a new network model with a rectilinear mesh2d
+    networkModel = NetworkModel()
+    networkModel.filepath = nc_output_file
+    network = networkModel.network
+    network.mesh2d_create_rectilinear_within_extent(extent=(-5, -5, 5, 5), dx=1, dy=1)
+
+    # set all values for the mesh2d_edge_z to NaN
```
Comment (arthurvd) [inline tests/io/test_net.py:576]: "_face_z"

File: tests/io/test_net.py [hunk 2]
```diff
@@ -559,3 +562,46 @@ def test_add_1d2d_links():
...
+    fill_value = 123.456
+    config = FillValueConfiguration()
+    config.float64_fill_value = fill_value
+    writer = UgridWriter(config)
+
+    writer.write(networkModel, nc_output_file)
+
+    # read the NetCDF we just wrote
+    dataset = nc.Dataset(nc_output_file)
+    mesh2d_edge_z = dataset["mesh2d_face_z"]
```
Comment (arthurvd) [inline tests/io/test_net.py:589]: edge / face mixup?

**Review by arthurvd — CHANGES_REQUESTED** [review]
Hi @tim-vd-aardweg : looks good overall, only two minor remarks that you can look at.

File: tests/io/test_net.py (hunk 1 above)
Comment (tim-vd-aardweg) [reply tests/io/test_net.py]: Good catch. Fixed.

File: tests/io/test_net.py (hunk 2 above)
Comment (tim-vd-aardweg) [reply tests/io/test_net.py]: Indeed. Fixed!

**Pull request merged**